src/Compilers/CSharp/Portable/Symbols/Source/SourceComplexParameterSymbol.cs

```diff
@@ -114,22 +114,19 @@ internal override bool IsIDispatchConstant
         internal override bool IsIUnknownConstant
             => GetDecodedWellKnownAttributeData()?.HasIUnknownConstantAttribute == true;
 
-        private bool HasCallerLineNumberAttribute
-            => GetEarlyDecodedWellKnownAttributeData()?.HasCallerLineNumberAttribute == true;
+        internal override bool IsCallerLineNumber => GetEarlyDecodedWellKnownAttributeData()?.HasCallerLineNumberAttribute == true;
 
-        private bool HasCallerFilePathAttribute
-            => GetEarlyDecodedWellKnownAttributeData()?.HasCallerFilePathAttribute == true;
+        internal override bool IsCallerFilePath => GetEarlyDecodedWellKnownAttributeData()?.HasCallerFilePathAttribute == true;
 
-        private bool HasCallerMemberNameAttribute
-            => GetEarlyDecodedWellKnownAttributeData()?.HasCallerMemberNameAttribute == true;
+        internal override bool IsCallerMemberName => GetEarlyDecodedWellKnownAttributeData()?.HasCallerMemberNameAttribute == true;
 
-        internal override bool IsCallerLineNumber => HasCallerLineNumberAttribute;
-
-        internal override bool IsCallerFilePath => !HasCallerLineNumberAttribute && HasCallerFilePathAttribute;
-
-        internal override bool IsCallerMemberName => !HasCallerLineNumberAttribute
-                                                  && !HasCallerFilePathAttribute
-                                                  && HasCallerMemberNameAttribute;
+        internal override int CallerArgumentExpressionParameterIndex
+        {
+            get
+            {
+                return GetEarlyDecodedWellKnownAttributeData()?.CallerArgumentExpressionParameterIndex ?? -1;
+            }
+        }
```

Comment:
Not sure if it makes sense to rename these from `Is*` to `Has*Attribute`. Thoughts @AlekseyTs @333fred ?

Reply:
> Not sure if it makes sense to rename these from Is* to Has*Attribute.

Given that PEParameterSymbol doesn't actually return a simple fact whether the attribute is applied, but also performs additional checks, I'd leave the names as is.

---
In reply to: [601886871](https://github.com/dotnet/roslyn/pull/51952#discussion_r601886871) [](ancestors = 601886871)